src/cfnlint/rules/parameters/Default.py

```diff
@@ -33,8 +33,11 @@ def check_allowed_pattern(self, allowed_value, allowed_pattern, path):
             Check allowed value against allowed pattern
         """
         message = 'Default should be allowed by AllowedPattern'
-        if not re.match(allowed_pattern, str(allowed_value)):
-            return([RuleMatch(path, message)])
+        try:
+            if not re.match(allowed_pattern, str(allowed_value)):
+                return([RuleMatch(path, message)])
+        except Exception:   # pylint: disable=W0703
```

Comment:
Can we use a more explicit Exception? And perhaps also output the error message in the logline?

Reply:
I think the ways it can fail result in different errors.  We found one error with the `multiple repeat` error.  We can capture them all as we run into them or all Re errors, maybe?

Reply:
Updated with something more useful... hopefully